samples/net/mqtt/src/modules/network/network_wifi.c

```diff
@@ -0,0 +1,165 @@
+		return;
+	}
+
+	LOG_INF("Connecting to SSID: %s", CONFIG_MQTT_SAMPLE_NETWORK_WIFI_SSID);
+
+	static struct wifi_connect_req_params cnx_params = {
+		.ssid = CONFIG_MQTT_SAMPLE_NETWORK_WIFI_SSID,
+		.ssid_length = sizeof(CONFIG_MQTT_SAMPLE_NETWORK_WIFI_SSID) - 1,
+		.psk = CONFIG_MQTT_SAMPLE_NETWORK_WIFI_PSK,
+		.psk_length = sizeof(CONFIG_MQTT_SAMPLE_NETWORK_WIFI_PSK) - 1,
+		.channel = WIFI_CHANNEL_ANY,
+		.security = WIFI_SECURITY_TYPE_PSK,
+		.timeout = CONFIG_MQTT_SAMPLE_NETWORK_WIFI_CONNECTION_REQUEST_TIMEOUT_SECONDS,
+	};
+
+	if (net_mgmt(NET_REQUEST_WIFI_CONNECT, iface,
+		     &cnx_params, sizeof(struct wifi_connect_req_params))) {
+		LOG_ERR("Connecting to Wi-Fi failed");
+	}
+}
+
+static void scan_result_handle(struct net_mgmt_event_callback *cb, uint32_t result_index)
+{
+	const struct wifi_scan_result *entry = (const struct wifi_scan_result *)cb->info;
+	uint8_t mac_buf[sizeof("xx:xx:xx:xx:xx:xx")];
+	const char *mac_ptr = (entry->mac_length) ? net_sprint_ll_addr_buf(entry->mac,
+									   WIFI_MAC_ADDR_LEN,
+									   mac_buf,
+									   sizeof(mac_buf)) : "";
+
+	if (result_index == 1U) {
+		LOG_INF("%-4s | %-32s %-5s | %-13s | %-4s | %-15s | %s",
+			"Num", "SSID", "(len)", "Chan (Band)", "RSSI", "Security", "BSSID");
+	}
+
+	LOG_INF("%-4d | %-32s %-5u | %-4u (%-6s) | %-4d | %-15s | %s",
+		result_index, entry->ssid, entry->ssid_length, entry->channel,
+		wifi_band_txt(entry->band),
+		entry->rssi,
+		wifi_security_txt(entry->security),
+		mac_ptr);
+}
+
+static void wifi_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+				    uint32_t mgmt_event, struct net_if *iface)
+{
+	int err;
+	enum network_status status;
+	const struct wifi_status *wifi_status = (const struct wifi_status *)cb->info;
+	/* Variable incremented for each new scan result. */
+	static uint32_t scan_result_index;
+
+	switch (mgmt_event) {
+	case NET_EVENT_WIFI_SCAN_RESULT:
+		scan_result_index++;
+		scan_result_handle(cb, scan_result_index);
+		return;
+	case NET_EVENT_WIFI_SCAN_DONE:
+		if (wifi_status->status) {
+			LOG_INF("Scan failed, status code: %d", wifi_status->status);
+			return;
+		}
+
+		LOG_INF("Scan finished");
+
+		scan_result_index = 0U;
+
+		connect();
+		return;
+	case NET_EVENT_WIFI_CONNECT_RESULT:
+		if (wifi_status->status) {
+			LOG_INF("Connection attempt failed, status code: %d", wifi_status->status);
+			return;
+		}
+
+		LOG_INF("Wi-Fi Connected");
+
+		status = NETWORK_CONNECTED;
+		break;
+	case NET_EVENT_WIFI_DISCONNECT_RESULT:
+		LOG_INF("Disconnected");
+
+		status = NETWORK_DISCONNECTED;
+		break;
+	default:
+		LOG_ERR("Unknown event: %d", mgmt_event);
+		return;
+	}
+
+	err = zbus_chan_pub(&NETWORK_CHAN, &status, K_SECONDS(1));
+	if (err) {
+		LOG_ERR("zbus_chan_pub, error: %d", err);
+		SEND_FATAL_ERROR();
+	}
+}
+
+static void network_task(void)
+{
+
+	net_mgmt_init_event_callback(&net_mgmt_callback, wifi_mgmt_event_handler, MGMT_EVENTS);
+	net_mgmt_add_event_callback(&net_mgmt_callback);
+
+	scan();
```

Comment:
I know that this implementation is temporary, but I wonder, do we really need to do the scanning, given that it has nothing to do with the acutal connection? I just have a feeling we unnecessarily bloat an already heavy sample.

Reply:
We can definitely trim the fat a bit by removing it 👍